File: Data Structures & Algorithms/two-integer-sum/submission-1.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} target
     * @return {number[]}
     */
    // Input: nums = [4,5,6], target = 10
    // Output: [0,2]

    twoSum(nums, target) {
        const numsObj = {};

        for(let i = 0; i < nums.length; i++) {
            const complement = target - nums[i];

            if(complement in numsObj){
                return [numsObj[complement], i];
            }

            numsObj[nums[i]] = i;
            console.log(`logging numsObj = ${JSON.stringify(numsObj)}`)
        }
    }
}

File: Data Structures & Algorithms/sliding-window-maximum/submission-1.py
class Solution:
    def maxSlidingWindow(self, nums: List[int], k: int) -> List[int]:
        output = []
        l = 0

        for r in range(k, len(nums) + 1): # O(n)
            output.append(max(nums[l:r])) # O(n)
            l+=1

        return output
